src/units/abilities.cpp

```diff
@@ -1969,6 +2151,15 @@ namespace
 			}
 		}
 
+		//when affect_distant=yes detect presence of [affect_distant] in abilities, if no
+		//then matches when tag not present.
+		if(!filter["affect_distant"].empty()){
+			bool distant = cfg.has_child("affect_distant");
+			if(filter["affect_distant"].to_bool() != distant){
+				return false;
+			}
+		}
```

Comment:
Is there a reason to support affect_distant as both an attribute and a subtag?

Reply:
It the same what for affect_adjacent,  verify what ability use or not this subtag, and it is used like attribute in filter_ability for check presence or absence of subtag ,if affect_distant=yes, abilities with subtag  match this critère, if equals not, abilties without subtag  matches

Reply:
Oh, right, got confused.